tiny_dnn/core/framework/tensor.h

```diff
@@ -108,7 +110,17 @@ class Tensor {
     const float_t& operator[] (cnn_size_t index) const {
         return *access_data(index);
     }
-    
+
+    void toDevice(const Device& device) {
+#if defined(USE_OPENCL) || defined(USE_CUDA)
+        CLCudaAPI::Context ctx = device.context();
+        CLCudaAPI::Queue queue = device.queue();
+
+        data_gpu_ = std::make_shared<CLCudaAPI::Buffer<float_t> >(
+            ctx, queue, data_cpu_->begin(), data_cpu_->end());
+#endif
+    }
+
```

Comment:
@nyanp  @bhack 
this is what I think that should be a minimal API to transfer data between devices.
It still needs more logic to handle corner cases. Do you have any consideration?